src/odbc/src/app/application_data_buffer.cpp

```diff
@@ -251,25 +255,52 @@ ConversionResult::Type ApplicationDataBuffer::PutStrToStrBuffer(
   LOG_DEBUG_MSG("inCharSize is " << inCharSize << ", outCharSize is "
                                  << outCharSize << ", buflen is " << buflen);
 
+  size_t bytesRequired = 0;
+  if (ANSI_STRING_ONLY) {
+    bytesRequired = value.length() * outCharSize;
+  } else {
+    static std::wstring_convert< std::codecvt_utf8< wchar_t >, wchar_t >
+      converter;
+    std::wstring inString = converter.from_bytes(value.c_str());
+    bytesRequired = inString.length() * outCharSize;
+  }
+
+  // Safety check since inCharSize and outCharSize will be denominators.
+  if (inCharSize == 0 || outCharSize == 0) {
+    LOG_ERROR_MSG(
+      "Unexpected conversion from unknown type string, inCharSize is "
+      << inCharSize << ", outCharSize is " << outCharSize);
+    assert(false);
```

Comment:
`outCharSize` will originate with the `type` in `ApplicationDataBuffer`. For `SQLGetData`, this will be the `targetType` value, after conversion with `ToDriverType`. `inCharSize` is internal.

The driver in general doesn't seem to throw many exceptions. In this case should I throw something?

Reply:
I think this if will never be reached as sizeof will never return 0 even for a empty class or struct.

Reply:
Good point. Removed.